sim/src/test/scala/bridges/BridgeSuite.scala

```diff
@@ -78,15 +72,69 @@ abstract class BridgeSuite(
   mkdirs()
   behavior.of(s"$targetName")
   elaborateAndCompile()
-  compileMlSimulator(backendSimulator)
-  runTest(backendSimulator)
+  for (backend <- Seq("vcs", "verilator")) {
+    compileMlSimulator(backend)
+    runTest(backend)
+  }
 }
 
+class UARTTest(targetConfig: String) extends BridgeSuite("UARTModule", "UARTConfig", targetConfig) {}
+
 class UARTF1Test    extends BridgeSuite("UARTModule", "UARTConfig", "DefaultF1Config")
 class UARTVitisTest extends BridgeSuite("UARTModule", "UARTConfig", "DefaultVitisConfig")
 
+class BlockDevTest(targetConfig: String) extends BridgeSuite("BlockDevModule", "BlockDevConfig", targetConfig) {
+  override def runTest(backend: String, debug: Boolean, shouldPass: Boolean) {
+    val prefix     = if (shouldPass) "pass in " else "fail in "
+    val testEnvStr = s"${backend} MIDAS-level simulation"
+    val wavesStr   = if (debug) " with waves enabled" else ""
+
+    val haveThisBehavior = prefix + testEnvStr + wavesStr
+
+    if (isCmdAvailable(backend)) {
+      // Generate a random string spanning 2 sectors with a fixed seed.
+      val gen = new scala.util.Random(100)
+
+      val alpha = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+      val data  = (1 to 1024).map(_ => alpha(gen.nextInt(alpha.length))).mkString
+
+      // Create an input file.
+      val input       = File.createTempFile("input", ".txt")
```

Comment:
This should be a separate method and reused to create both files.

Reply:
See #1336